game_mechanics.py

```diff
@@ -79,23 +81,119 @@ def _make_move(board: np.ndarray, move: Tuple[int, int], current_player: int) ->
 def is_legal_move(board: np.ndarray, move: Tuple[int, int], current_player: int) -> bool:
     board_dim = board.shape[0]
     if is_valid_coord(board_dim, move[0], move[1]) and board[move] == 0:
-        for direction in MOVE_DIRS:
-            if has_tile_to_flip(board, move, direction, current_player):
-                return True
+        if has_tile_to_flip(board, move, current_player):
+            return True
     return False
```

Comment:
Well I'm not allowed to suggest anything here by GitHub because I'm over my quota (real reason is you can't suggest over deleted lines)

...but you know what I _would_ suggest :wink: